sklearn/utils/estimator_checks.py

```diff
@@ -252,6 +241,21 @@ def _check_transformer(name, Transformer, X, y):
             assert_raises(ValueError, transformer.transform, X.T)
 
 
+def check_fit_score_takes_y(name, Estimator):
+    # check that all estimators accept an optional y
+    # in fit and score so they can be used in pipelines
+    rnd = np.random.RandomState(0)
+    X = rnd.uniform(size=(10, 3))
+    y = (X[:, 0] * 4).astype(np.int)
+    y = multioutput_estimator_convert_y_2d(name, y)
+    with warnings.catch_warnings(record=True):
+        estimator = Estimator()
+    set_fast_parameters(estimator)
+    set_random_state(estimator)
+    estimator.fit(X, y)
```

Comment:
Hmm. We're checking here whether the method takes more than one positional argument. Do we rather need a whitelist of argument names for the second arg so that we're not inadvertently setting `sample_weight`?

Also, do we need a parallel check for `partial_fit`, even if not currently used in `Pipeline`? Certainly it would simplify [this code](https://github.com/scikit-learn/scikit-learn/pull/3907/files#diff-a95fe0e40350c536a5e303e87ac979c4R386)